File: Data Structures & Algorithms/split-array-largest-sum/submission-0.py
class Solution:
    def splitArray(self, nums: List[int], k: int) -> int:
        max_val=0
        sum_n=0
        n=len(nums)
        for e in nums:
            max_val=max(max_val,e)
            sum_n+=e
        low=max_val
        high=sum_n
        
        while low < high :
            mid=(low+high)//2
            splits=1
            curr_sum=0
            for e in nums:
                if curr_sum + e <= mid :
                    curr_sum+=e
                    continue
                else : 
                    splits+=1
                    curr_sum=e
                    

            if splits <= k :  high = mid
            else : low = mid + 1
        
        return low
                

        

File: Data Structures & Algorithms/search-insert-position/submission-0.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        // search space -> All indices of the array from 0 to num
        int n=nums.size();
        int low=0;
        int high=n-1;
        if(target > nums[n-1]) return n;
        while(low < high) {
            int mid = (low + high) / 2 ;
            // feasibilty condition 
            if(nums[mid]>=target) {
                high = mid;
            }
            else {
                low = mid + 1;
            }

        }
        return low;
    }
};